presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Rule.java

```diff
@@ -46,4 +48,40 @@ default boolean isEnabled(Session session)
 
         Session getSession();
     }
+
+    class Result
+    {
+        @Deprecated
+        public static Result fromOptional(Optional<PlanNode> transformedPlan)
+        {
+            return new Result(transformedPlan);
+        }
+
+        public static Result empty()
+        {
+            return new Result(Optional.empty());
+        }
+
+        public static Result replace(PlanNode transformedPlan)
+        {
+            return new Result(Optional.of(transformedPlan));
+        }
+
+        private final Optional<PlanNode> transformedPlan;
+
+        private Result(Optional<PlanNode> transformedPlan)
+        {
+            this.transformedPlan = requireNonNull(transformedPlan, "transformedPlan is null");
+        }
+
+        public Optional<PlanNode> getTransformedPlan()
+        {
+            return transformedPlan;
+        }
+
+        public boolean isPresent()
```

Comment:
As commented in #8852, maybe we can live without this method?
Now it's clear what it means, but as soon as you add traits here, it will no longer be.

Reply:
+1, in any case, I would make it more explicit, e.g: `isTraitPresent` or `isPlanNodePresent`?